toolchain/semantics/semantics_handle_index.cpp

```diff
@@ -5,17 +5,56 @@
 #include <utility>
 
 #include "toolchain/semantics/semantics_context.h"
+#include "toolchain/semantics/semantics_node.h"
+#include "toolchain/semantics/semantics_node_kind.h"
 
 namespace Carbon {
 
 auto SemanticsHandleIndexExpressionStart(SemanticsContext& context,
                                          ParseTree::Node parse_node) -> bool {
-  return context.TODO(parse_node, "HandleIndexExpressionStart");
+  context.node_stack().Push(parse_node, context.node_stack().PopExpression());
```

Comment:
```suggestion
  // Leave the expression on the stack for IndexExpression.
```

We've been trying an approach of minimizing push/pop. Since the IndexExpressionStart isn't significant, we don't need to add it.